### ISSUE TITLE
NPE if global setting "implicit.host.tags" is set to null

### ISSUE THREAD

**Issue by weizhouapache** [apache/cloudstack#8056]
steps to reproduce the issue
- add xen hosts with GPU, this can be simularted by database change
```
insert into host_gpu_groups select 1,'GPU',id from host where type='Routing' and removed is null limit 1;
```
- change global setting "implicit.host.tags" to null
- deploy a new vm

Got an exception:
```
2023-10-09 10:11:22,381 ERROR [c.c.a.ApiAsyncJobDispatcher] (API-Job-Executor-8:ctx-5c18d0af job-1130) (logid:815ab064) Unexpected exception while executing org.apache.cloudstack.api.command.admin.vm.DeployVMCmdByAdmin
java.lang.NullPointerException
        at com.cloud.deploy.FirstFitPlanner.reorderClustersBasedOnImplicitTags(FirstFitPlanner.java:218)
        at com.cloud.deploy.FirstFitPlanner.orderClusters(FirstFitPlanner.java:206)
        at com.cloud.deploy.DeploymentPlanningManagerImpl.planDeployment(DeploymentPlanningManagerImpl.java:568)
```

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~

~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by weizhouapache** [comment]
this issue may be fixed by
```
diff --git a/server/src/main/java/com/cloud/deploy/FirstFitPlanner.java b/server/src/main/java/com/cloud/deploy/FirstFitPlanner.java
index 695521e5272..4e838510010 100644
--- a/server/src/main/java/com/cloud/deploy/FirstFitPlanner.java
+++ b/server/src/main/java/com/cloud/deploy/FirstFitPlanner.java
@@ -124,7 +124,7 @@ public class FirstFitPlanner extends AdapterBase implements DeploymentClusterPla
 
     protected String allocationAlgorithm = "random";
     protected String globalDeploymentPlanner = "FirstFitPlanner";
-    protected String[] implicitHostTags;
+    protected String[] implicitHostTags = new String[0];
 
```

**Comment by harikrishna-patnala** [comment]
yes @weizhouapache fix makes sense, let me test this and raise a PR or do any other changes required. Thanks

**Comment by shwstppr** [comment]
Fixed by #8066
